test/upgrade/service_postupgrade_test.go

```diff
@@ -62,6 +72,30 @@ func TestBYORevisionPostUpgrade(t *testing.T) {
 	}
 }
 
+func configHasGeneration(t *testing.T, clients *test.Clients, serviceName string, generation int64) bool {
```

Comment:
I think you want this return condition reversed. Return true if the generation matches and false if it doesn't